FreeRTOS/FreeRTOS
# Add Renesas RX72N Envision Kit RTOS Demo using Renesas RXv3 port layer

Description
-----------
As I told in the pull request https://github.com/FreeRTOS/FreeRTOS-Kernel/pull/104,  I'd like to post a pull request to add Renesas RX72N Envision Kit RTOS Demo using Renesas RXv3 port layer. Build environment, download pages, screen snap shots, etc are written/uploaded in the following page.

[https://github.com/NoMaY-jp/FreeRTOS/wiki/RX72N-Envision-Kit-RTOS-Demo-Programs](https://github.com/NoMaY-jp/FreeRTOS/wiki/RX72N-Envision-Kit-RTOS-Demo-Programs)

Almost code are generated code by Renesas RX SmartConfigurator so that folder structure is different from other Renesas RX RTOS Demo projects. Moreover, in case of FreeRTOS project which is generated by e2 studio, generated FreeRTOS project does not have main() so that the following definition is placed in src/FreeRTOS_Demo/demo_main.h
```
/* Set mainCREATE_SIMPLE_BLINKY_DEMO_ONLY to one to run the simple blinky demo,
or 0 to run the more comprehensive test and demo application. */
#define mainCREATE_SIMPLE_BLINKY_DEMO_ONLY  0
```
Test Steps
-----------
(A) Simple Blinky Demo and Full Demo are checked using using on-board LED with the following 5 cases.

e2 studio + GNURX, e2 studio + CC-RX, e2 studio + ICCRX, CS+ for CC (CC-RX), EWRX (ICCRX)

(B) Full Demo's console input/output are checked using terminal software with the same following 5 cases.

e2 studio + GNURX, e2 studio + CC-RX, e2 studio + ICCRX, CS+ for CC (CC-RX), EWRX (ICCRX)

Both (A) and (B) use the following setting.

($-1) Not define USE_FULL_REGISTER_INITIALISATION
($-2) Define configUSE_TASK_DPFPU_SUPPORT = 1

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
